app/services/label_templates_preview_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class LabelTemplatesPreviewService
+  extend Service
+
+  attr_reader :error, :preview
+
+  def initialize(label, user, params)
```

Comment:
Could this be change to accept ZPL directly? As we have preview on the template page, and then also on the actual printing page, so in that case we accept the already rendered ZPL, which is not what's in the content part of a label.